segmentation/utils/segmentation_utils.py

```diff
@@ -297,103 +301,92 @@ def segment_images(input_images, segmentation_masks):
     unique_cell_num = len(np.unique(segmentation_masks.values).astype('int'))
 
     # create np.array to hold subcellular_loc x channel x cell info
-    cell_counts = np.zeros((len(segmentation_masks.subcell_loc), unique_cell_num,
+    cell_counts = np.zeros((len(segmentation_masks.compartments), unique_cell_num,
                             len(input_images.channels) + 1))
 
     col_names = np.concatenate((np.array('cell_size'), input_images.channels), axis=None)
     xr_counts = xr.DataArray(copy.copy(cell_counts),
-                             coords=[segmentation_masks.subcell_loc,
+                             coords=[segmentation_masks.compartments,
                                      np.unique(segmentation_masks.values).astype('int'), col_names],
-                             dims=['subcell_loc', 'cell_id', 'features'])
+                             dims=['compartments', 'cell_id', 'features'])
+
+    # get regionprops for each cell
+    cell_props = regionprops(segmentation_masks.loc[:, :, 'whole_cell'].values)
+
+    # loop through each cell in mask
+    for cell in cell_props:
+        # get coords corresponding to current cell.
+        # TODO: This is faster than mask-based indexing, but leads to confusing code. Is there
+        # TODO a better way to use mask-based indexing to extract counts?
```

Comment:
Not necessarily indexing, but if there's a good way of getting a mask as a `scipy.sparse.dok_matrix`, point-wise multiplying it by the image and calling sum on the resultant sparse matrix should be super fast and possibly more understandable at-a-glance.